src/modules/assisted_release/assisted_release.cpp

```diff
@@ -0,0 +1,312 @@
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include "assisted_release.hpp"
+
+#include <px4_platform_common/getopt.h>
+#include <px4_platform_common/log.h>
+#include <px4_platform_common/posix.h>
+
+#include <uORB/Publication.hpp>
+#include <drivers/drv_hrt.h>
+
+#include <uORB/topics/parameter_update.h>
+#include <uORB/topics/sensor_combined.h>
+
+#include <uORB/topics/vehicle_status.h>
+#include <uORB/topics/actuator_controls.h>
+#include <uORB/topics/airspeed.h>
+#include <uORB/topics/rpm.h>
+
+
+int AssistedRelease::print_status()
+{
+	PX4_INFO("Running");
+    PX4_INFO("State: frequency=%d, airspeed=%d, throttle=%d, switch=%d", _rpm.indicated_frequency_rpm > (float)_param_min_rpm.get(), _airspeed.indicated_airspeed_m_s > (float)_param_min_aspd.get(), _throttle >= 0.5f, _rc_channel >= 0.5f);
+
+	return 0;
+}
+
+int AssistedRelease::custom_command(int argc, char *argv[])
+{
+	/*
+	if (!is_running()) {
+		print_usage("not running");
+		return 1;
+	}
+
+	// additional custom commands can be handled like this:
+	if (!strcmp(argv[0], "do-something")) {
+		get_instance()->do_something();
+		return 0;
+	}
+	 */
+
+	return print_usage("unknown command");
+}
+
+
+int AssistedRelease::task_spawn(int argc, char *argv[])
+{
+	_task_id = px4_task_spawn_cmd("assisted_release",
+				      SCHED_DEFAULT,
+				      SCHED_PRIORITY_DEFAULT,
+				      1024,
+				      (px4_main_t)&run_trampoline,
+				      (char *const *)argv);
+
+	if (_task_id < 0) {
+		_task_id = -1;
+		return -errno;
+	}
+
+	return 0;
+}
+
+AssistedRelease *AssistedRelease::instantiate(int argc, char *argv[])
+{
+	int example_param = 0;
+	bool example_flag = false;
+
+	AssistedRelease *instance = new AssistedRelease(example_param, example_flag);
+
+	if (instance == nullptr) {
+		PX4_ERR("alloc failed");
+	}
+
+	return instance;
+}
+
+AssistedRelease::AssistedRelease(int example_param, bool example_flag)
+	: ModuleParams(nullptr)
+{
+}
+
+void AssistedRelease::run()
+{
+
+	// int parameter_update_sub = orb_subscribe(ORB_ID(parameter_update));
+
+	_rpm_sub = orb_subscribe(ORB_ID(rpm));
+	_vehicle_status_sub = orb_subscribe(ORB_ID(vehicle_status));
+	_airspeed_sub = orb_subscribe(ORB_ID(airspeed));
+	_actuator_controls_0_sub = orb_subscribe(ORB_ID(actuator_controls_0));
+	_input_rc_sub = orb_subscribe(ORB_ID(input_rc));
+
+	orb_set_interval(_rpm_sub, 100);
+	orb_set_interval(_vehicle_status_sub, 100);
+	orb_set_interval(_airspeed_sub, 100);
+	orb_set_interval(_actuator_controls_0_sub, 100);
+	orb_set_interval(_input_rc_sub, 100);
+
+	rpm_poll();
+	airspeed_poll();
+	vehicle_status_poll();
+	actuator_controls_poll();
+	input_rc_poll();
+
+	actuator_controls_s control{};
+	uORB::Publication<actuator_controls_s> control_pub{ORB_ID(actuator_controls_1)};
+
+
+	px4_pollfd_struct_t fds[] = {
+	    { .fd = _rpm_sub,   .events = POLLIN },
```

Comment:
What's the intention here? I see below you pass in 1 to px4_poll(), which means you're only actually polling `rpm`. If you actually want to poll all these topics it might be worth discussing, because that usually leads to things running erratically.